Add support for map to clues

deathbeam opened the pull request:
## Add instance map service to open map at location

- Add service to instance map plugin that will open map at specified
location
- Use new world map drawer in instance map to draw different regions

## Add support for opening map with location clues

Open world map from location clue:
1. Automatically when clue is first read
2. When check-steps on clue is called

Depends on: #1631 
Signed-off-by: Tomas Slusny <slusnucky@gmail.com>

Preview:
![peek 2018-04-11 14-46](https://user-images.githubusercontent.com/5115805/38617632-d6b8398c-3d97-11e8-8b8a-df83affd185a.gif)

deathbeam commented:
By skipping all "empty" (e.g. black regions) during dumping and then distributing in-memory fully black region when dump is not found I managed to reduce the file size of world map from 28MB to 13MB.

```
$ \du -kh runelite-map/target/classes/runelite/map/
13M     runelite-map/target/classes/runelite/map/
```

MESLewis commented:
Some feedback as a user

Displaying on first read is a big jarring
Ideally the map could have a X to close.
"Check Steps" could toggle instead of just showing the map.

Right now the only ways to close the map are ESC or right clicking minimap orb

deathbeam commented:
1. I think people wanted to display it on first read to work same as OSB.
2. The hiding is more of an instance map problem, that should probably be done during the world map changes
3. Hmm yeah, I could easily do that, that would be nice I guess

MESLewis commented on runelite-client/src/main/java/net/runelite/client/plugins/cluescrolls/ClueScrollPlugin.java:
```diff
@@ -201,6 +219,12 @@ public void onGameTick(final GameTick event)
+				// Set hint arrow to first object found as there can only be 1 hint arrow
+				if (objectsToMark.length >= 1)
+				{
+					client.setHintArrow(objectsToMark[0].getWorldLocation());
```
A check can be added to loop through objectsToMark to find an object location that matches the clue.getLocation()

deathbeam replied:
This comment should be probably on #1631

MESLewis replied:
I'll move it there. Didn't realize this was from a dependency.

Nightfirecat commented:
OSBuddy doesn't have "display on first read"; it uses a minimap orb that appears on the clue interface.

deathbeam commented:
People were saying that on OSB it auto-shows. If that is not the case, I can remove that behaviour.

MagicfTail commented:
I'm pretty sure osbuddy does open on first read, and even if it doesn't it's really nice since you're going to want to open the map anyways in 99% of all cases

deathbeam commented:
1. Added configurable behaviour for opening the map on first read.
2. Split the world map assembler commit to 2 commits, 1 for updating the cache with exposes StoreLocation, and second one for adding the map assembler itself so it can be split to separate repo at any point.

deathbeam commented:
Closing in favour of #2064
